boto3/dynamodb/types.py

```diff
@@ -62,6 +62,10 @@ def __repr__(self):
         return 'Binary(%r)' % self.value
 
     def __str__(self):
+        return self.value if isinstance(self.value, six.string_types) \
+            else self.value.decode('utf-8')
```

Comment:
I don't think this is a safe assumption.  Given `self.value` can be any arbitrary binary data, we can't assume we can decode this properly as utf-8.

```
In [6]: str(Binary(os.urandom(10)))
---------------------------------------------------------------------------
UnicodeDecodeError                        Traceback (most recent call last)
<ipython-input-6-e9eb5b72f2e9> in <module>()
----> 1 str(Binary(os.urandom(10)))

/boto3/boto3/dynamodb/types.py in __str__(self)
     64     def __str__(self):
     65         return self.value if isinstance(self.value, six.string_types) \
---> 66             else self.value.decode('utf-8')
     67
     68     def __bytes__(self):

UnicodeDecodeError: 'utf-8' codec can't decode byte 0xb7 in position 0: invalid start byte
```